core/src/main/java/com/oath/oak/Handle.java

```diff
@@ -196,27 +200,33 @@ public ByteOrder order() {
         return transformation;
     }
 
-    /**
-     * Applies a transformation under writers locking
-     *
-     * @param transformer transformation to apply
-     * @return Transformation result or null if value is deleted
-     */
-    <T> T mutatingTransform(Function<ByteBuffer, T> transformer) {
-        T result;
+    <V> V exchange(V newValue, Function<ByteBuffer, V> valueDeserializeTransformer, OakSerializer<V> serializer, MemoryManager memoryManager) {
         try {
             writeLock.lock();
-            if (isDeleted()) {
-                // finally clause will handle unlock
+            if (isDeleted())
                 return null;
-            }
-            result = transformer.apply(getSlicedByteBuffer());
+            V v = valueDeserializeTransformer.apply(this.value);
```

Comment:
so this is the core of the change bug fix (!)
here innerPut does not take a lock where previously `transformer.apply` used a lock
somewhat hard to notice this in the diff-clutter github created.

maybe a comment documenting `innerPut` and specifically mentioning it assumes a lock is taken by the invoking method can help